server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java

```diff
@@ -1371,4 +1371,12 @@ public void testParsingMissingTimezone() {
         long millisJoda = DateFormatter.forPattern("yyyy-MM-dd HH:mm:ss").parseMillis("2018-02-18 17:47:17");
         assertThat(millisJava, is(millisJoda));
     }
+
+    // see https://bugs.openjdk.org/browse/JDK-8193877
+    public void testNoClassCastException() {
+        String input = "DpNKOGqhjZ";
```

Comment:
Shouldn't we randomise this? Or does the bug only manifest with some input format and this is such an instance? (We should add a comment if such.)

Reply:
It happens only for some specific patterns, I linked the JDK bug as it explains the problem much better than how I could do it in a comment